### ISSUE TITLE
Allow for users to specify which OPENAI model

### ISSUE THREAD

**Issue by thomasdavis** [IvanIsCoding/ResuLLMe#23]
and/or just set the default to `gpt-4o` 

Creator of JSON Resume here, I've been building out AI tools for the project and `gpt-4o` is far superior. (cheap as chips too) 

I don't know how I didn't see this project, it is super awesome, lots of cool ideas. (added it to official projects page)

**Comment by IvanIsCoding** [comment]
I will add a toggle to choose the OpenAI model, I think that is a great idea.

When I originally impelemented ResuLLMe, GPT4 had just been released and it was somewhat expensive. So I defaulted to ChatGPT at the time.

I might make GPT4 the default depending on how good gpt4o is at generating JSON. Ideally we'd use more of the new features to generate JSON like function calling does. But at least for ChatGPT the prompting got better results when I exploited TypeScript-based prompts.
